### PR TITLE
Add arch flag to assembler to fix build on M1 mac.

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/pull/59167#issuecomment-1172047037

### CODE REVIEW

File: platform/osx/detect.py [hunk 1]
```diff
@@ -79,10 +79,12 @@ def configure(env):
 
     if env["arch"] == "arm64":
         print("Building for macOS 11.0+, platform arm64.")
+        env.Append(ASFLAGS=["-arch", "arm64", "-mmacosx-version-min=11.0"])
         env.Append(CCFLAGS=["-arch", "arm64", "-mmacosx-version-min=11.0"])
         env.Append(LINKFLAGS=["-arch", "arm64", "-mmacosx-version-min=11.0"])
     else:
         print("Building for macOS 10.12+, platform x86_64.")
+        env.Append(ASFLAGS=["-arch", "x86_64", "-mmacosx-version-min=11.0"])
```
Comment (akien-mga) [inline platform/osx/detect.py:87]: Shouldn't this be 10.2?

Comment (bruvzg) [reply platform/osx/detect.py]: Should be `10.12`, fixed.

**Comment by akien-mga** [comment]
You could possibly add the same to the iOS build, as I ran into the same issue there while testing the PR.

I then excluded iOS from the list but it could likely be added back:
https://github.com/godotengine/godot/blob/110248cd57b5abb82c216d4a117fd7696a593931/core/SCsub#L132L135

That would only impact the simulator build though and maybe it's best if it's not used there, so we don't have unexpected differences between real hardware and simulator. (But fixing ASFLAGS would still be good IMO.)

**Comment by bruvzg** [comment]
Add the same for iOS, and enabled `huf_decompress_amd64.S` build for iOS.

Note for cherry-picking: iOS SDK versions are different in the 3.x branch (10.0 for both device and simulator).

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.6.

**Pull request merged**